src/models/StyledNativeComponent.js

```diff
@@ -107,7 +123,7 @@ export default (constructWithOptions: Function) => {
       }
 
       if (!isStyledComponent(target)) {
-        propsForElement.ref = innerRef
+        propsForElement.ref = this.generateRef()
```

Comment:
The `setNativeProps` method needs to work for nested Styled Components too, so we need to add a case for passing `innerRef` as well.